include/private/dns/system_resolver.hpp

```diff
@@ -45,20 +45,32 @@ void system_resolver(QueryClass dns_class, QueryType dns_type, std::string name,
         return;
     }
 
+    /*
+     * When running OONI tests, we're interested to know not only the IPs
+     * associated with a specific name, but also to the CNAME.
+     */
+    ErrorOr<bool> also_cname = settings.get("dns/resolve_also_cname", false);
```

Comment:
I am not a fan of this settings name, but I understand why you used it (`resolve_cname` would have been confusing). No concrete suggestion here, but mainly a note to myself.